src/main/java/org/openrewrite/java/testing/mockito/MockitoJUnitToMockitoExtension.java

```diff
@@ -0,0 +1,219 @@
+ * Refactoring the method may be fairly complex and would likely benefit from being a separate recipe.
+ */
+public class MockitoJUnitToMockitoExtension extends Recipe {
+    private static final ThreadLocal<JavaParser> JAVA_PARSER = ThreadLocal.withInitial(() ->
+            JavaParser.fromJavaVersion()
+                    .dependsOn(Arrays.asList(
+                            Parser.Input.fromString("package org.mockito.junit;\n" +
+                                    "public class MockitoJUnit {\n" +
+                                        "public static MockitoRule rule() { return null; }\n" +
+                                        "public static MockitoTestRule testRule(Object testInstance) { return null; }\n" +
+                                    "}"),
+                            Parser.Input.fromString("package org.mockito.junit;\n" +
+                                    "public interface MockitoRule {\n" +
+                                    "  MockitoRule silent();\n" +
+                                    "}"),
+                            Parser.Input.fromString("package org.mockito.junit;\n" +
+                                    "public class MockitoJUnit {\n" +
+                                    "  MockitoRule rule() { return null; };\n" +
+                                    "  MockitoTestRule testRule() { return null; };\n" +
+                                    "  VerificationCollector collector() { return null; };\n" +
+                                    "}"),
+                            Parser.Input.fromString("package org.junit.jupiter.api.extension;\n" +
+                                    "public @interface ExtendWith {\n" +
+                                    "  Class[] value();\n" +
+                                    "}"),
+                            Parser.Input.fromString("package org.mockito.junit.jupiter;\n" +
+                                    "public class MockitoExtension {\n" +
+                                    "}")
+                    ))
+                    .build()
+    );
+
+    @Override
+    public String getDisplayName() {
+        return "MockitoJUnit to MockitoExtension";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces MockitoJUnit rules with MockitoExtension.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MockitoRuleToMockitoExtensionVisitor();
+    }
+
+    public static class MockitoRuleToMockitoExtensionVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private static final String MOCKITO_RULE_FQN = "org.mockito.junit.MockitoRule";
+        private static final String MOCKITO_TEST_RULE_FQN = "org.mockito.junit.MockitoTestRule";
+
+        private static final String MOCKITO_RULE_INVOCATION_KEY = "mockitoRuleInvocation";
+        private static final String MOCKITO_TEST_RULE_INVOCATION_KEY = "mockitoTestRuleInvocation";
+
+        @SneakyThrows
```

Comment:
As much as I dislike checked exceptions I'm not sure about this. If surrounding code were to try and catch a checked exception that was thrown "sneakily" IDEs and the like would put up spurious warnings about catching it. What checked exception is being thrown in this case?

Reply:
I snaked this concept from a recipe Tyler wrote, the new RuleBasedCollator("< ExtendWith") has a checked java.text.ParseException.

I'll ask him about it tomorrow, and see if there's anything I can do to refactor the code.